crates/red_knot_python_semantic/resources/mdtest/call/union.md

```diff
@@ -162,6 +162,30 @@ def _(flag: bool):
     reveal_type(f("string"))  # revealed: Literal["string", "'string'"]
 ```
 
+## Unions with literals and negations
+
+```py
+from typing import Literal, Union
+from knot_extensions import Not, AlwaysFalsy, static_assert, is_subtype_of, is_assignable_to
+
+static_assert(is_subtype_of(Literal["a", ""], Union[Literal["a", ""], Not[AlwaysFalsy]]))
+static_assert(is_subtype_of(Not[AlwaysFalsy], Union[Literal["", "a"], Not[AlwaysFalsy]]))
+static_assert(is_subtype_of(Literal["a", ""], Union[Not[AlwaysFalsy], Literal["a", ""]]))
+static_assert(is_subtype_of(Not[AlwaysFalsy], Union[Not[AlwaysFalsy], Literal["a", ""]]))
+
+static_assert(is_subtype_of(Literal["a", ""], Union[Literal["a", ""], Not[Literal[""]]]))
+static_assert(is_subtype_of(Not[Literal[""]], Union[Literal["a", ""], Not[Literal[""]]]))
+static_assert(is_subtype_of(Literal["a", ""], Union[Not[Literal[""]], Literal["a", ""]]))
+static_assert(is_subtype_of(Not[Literal[""]], Union[Not[Literal[""]], Literal["a", ""]]))
+
+def _(
+    x: Union[Literal["a", ""], Not[AlwaysFalsy]],
+    y: Union[Literal["a", ""], Not[Literal[""]]],
+):
+    reveal_type(x)  # revealed: Literal[""] | ~AlwaysFalsy
+    reveal_type(y)  # revealed: Literal[""] | ~Literal[""]
```

Comment:
shouldn't this simplify to `object`? Doesn't need to be done in this PR of course, but could add a TODO?

Reply:
Looks like this is also a regression from the unions-of-literals optimization; I won't fix it in this PR but I will fix it.